src/components/nav/index.js

```diff
@@ -55,6 +55,69 @@ const Nav = () => {
     }
   }, [joinedSpaces, auth.user, matrixClient, spacesErr])
 
+  useEffect(() => {
+    async function checkRoomForPossibleInvite (room) {
+      // Types of spaces for which we want to show invites
+      const typesOfSpaces = [
+        'studentproject',
+        'context',
+        'class',
+        'course',
+        'institution',
+        'degree program',
+        'design department',
+        'faculty',
+        'institute',
+        'semester']
+
+      // Ignore if this is not a space
+      if (room.getType() !== 'm.space') return
+      // Ignore if this is not a student project or a "context"
+      const metaEvent = await matrixClient.getStateEvent(room.roomId, 'dev.medienhaus.meta').catch(() => { })
+      if (!metaEvent || !metaEvent.type || !typesOfSpaces.includes(metaEvent.type)) return
+      // Ignore if this is not an invitation (getMyMembership() only works correctly after calling _loadMembersFromServer())
+      await room.loadMembersFromServer().catch(console.error)
+      if (room.getMyMembership() !== 'invite') return
+      // At this point we're sure that this is an invitation we want to display, so we add it to the state:
+      setInvites((invites) => [...invites, room.roomId])
+    }
+
+    async function removeInvites (room) {
+      const typesOfSpaces = [
+        'studentproject',
+        'context',
+        'class',
+        'course',
+        'institution',
+        'degree program',
+        'design department',
+        'faculty',
+        'institute',
+        'semester']
+
+      // Ignore if this is not a space
+      if (room.getType() !== 'm.space') return
+      // Ignore if this is not a student project or a "context"
+      const metaEvent = await matrixClient.getStateEvent(room.roomId, 'dev.medienhaus.meta').catch(() => { })
+      if (!metaEvent || !metaEvent.type || !typesOfSpaces.includes(metaEvent.type)) return
+      if (room.selfMembership === 'invite') return // we want invites to stay in the array
+      setInvites(invites => invites.filter(invite => invite !== room.roomId))
```

Comment:
Because effectively the only difference is at the end: where want to either -add- the room ID to our `invites` state, or where we want to -remove- the room ID from it. (or at least make sure that it's not in there)

So we only need 1 function, which we can still call `checkRoomForPossibleInvite`, but pay attention to the last `if/else`:
```js
    async function checkRoomForPossibleInvite (room) {
      // Types of spaces for which we want to count invites for
      const typesOfSpaces = [
        'studentproject',
        'context',
        'class',
        'course',
        'institution',
        'degree program',
        'design department',
        'faculty',
        'institute',
        'semester']

      // Ignore if this is not a space
      if (room.getType() !== 'm.space') return
      // Ignore if this is not a student project or a "context"
      const metaEvent = await matrixClient.getStateEvent(room.roomId, 'dev.medienhaus.meta').catch(() => { })
      if (!metaEvent || !metaEvent.type || !typesOfSpaces.includes(metaEvent.type)) return
      // Ignore if this is not an invitation (getMyMembership() only works correctly after calling _loadMembersFromServer())
      await room.loadMembersFromServer().catch(console.error)
      // At this point we're sure that the room we're checking for is either
      if (room.getMyMembership() === 'invite') {
        // ... 1. an invitation we want to display, so we add it to the state, or ...
        setInvites((invites) => [...invites, room.roomId])
      } else {
        // ... 2. a room that the membership has changed for to something other than "invite", so we do -not- want it to be in the state in case it's there right now:
        setInvites(invites => invites.filter(invite => invite !== room.roomId))
      }
    }
```

And then we only subscribe/unsubscribe to that one function below.